pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -179,6 +176,29 @@ def reduce_early_stopping_decision(self, should_stop: bool) -> bool:
         should_stop = int(stop.item()) == self.world_size
         return should_stop
 
+    def reduce(self, output, group: Optional[Any] = None, reduce_op: Optional[Union[ReduceOp, str]] = None):
+        if not isinstance(output, torch.Tensor):
+            output = torch.tensor(output, device=self.device)
+
+        if (isinstance(reduce_op, ReduceOp) and ReduceOp != ReduceOp.SUM) \
+           or isinstance(reduce_op, str) and reduce_op.lower() not in ("sum", "mean", "avg"):
```

Comment:
```suggestion
        _valid_reduce_op_str = isinstance(reduce_op, str) and reduce_op.lower() not in ("sum", "mean", "avg")
        if (isinstance(reduce_op, ReduceOp) and reduce_op != ReduceOp.SUM) or _valid_reduce_op_str:
```
but what is this one `ReduceOp != ReduceOp.SUM`? shall it be `reduce_op != ReduceOp.SUM`